File: frontend/src/api/project.ts
import request from './request'

// 章节摘要（项目列表中嵌套）
export interface ChapterSummary {
  id: number
  title: string
  sort_order: number
  word_count: number
  status: string
  pov_character: string | null
  created_at: string
  updated_at: string | null
}

// 项目数据类型
export interface Project {
  id: number
  title: string
  description: string
  genre: string
  target_word_count: number
  current_word_count: number
  status: string
  outline: string
  created_at: string
  updated_at: string
  chapters: ChapterSummary[]
}

export interface CreateProjectData {
  title: string
  description?: string
  genre?: string
  target_word_count?: number
  outline?: string
}

export interface UpdateProjectData {
  title?: string
  description?: string
  genre?: string
  target_word_count?: number
  current_word_count?: number
  status?: string
  outline?: string
}

// 获取项目列表
export async function getProjects(status?: string): Promise<Project[]> {
  return request.get<Project[]>('/projects/', {
    params: status ? { status } : undefined,
  })
}

// 创建项目
export async function createProject(data: CreateProjectData): Promise<Project> {
  return request.post<Project>('/projects/', data)
}

// 获取单个项目
export async function getProject(id: number): Promise<Project> {
  return request.get<Project>(`/projects/${id}/`)
}

// 更新项目
export async function updateProject(id: number, data: UpdateProjectData): Promise<Project> {
  return request.put<Project>(`/projects/${id}/`, data)
}

// 删除项目
export async function deleteProject(id: number): Promise<void> {
  return request.delete(`/projects/${id}/`)
}

File: backend/app/schemas/project.py
"""
项目 Pydantic 模式
"""
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, ConfigDict


class ProjectCreate(BaseModel):
    """创建项目时的请求体"""
    title: str
    description: Optional[str] = None
    genre: Optional[str] = None
    target_word_count: int = 0
    status: str = "draft"
    outline: Optional[str] = None


class ProjectUpdate(BaseModel):
    """更新项目时的请求体（所有字段可选）"""
    title: Optional[str] = None
    description: Optional[str] = None
    genre: Optional[str] = None
    target_word_count: Optional[int] = None
    current_word_count: Optional[int] = None
    status: Optional[str] = None
    outline: Optional[str] = None


class ChapterSummary(BaseModel):
    """章节简要信息（嵌入在项目响应中）"""
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    sort_order: int
    word_count: int
    status: str
    pov_character: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None


class ProjectResponse(BaseModel):
    """项目详情响应（含章节列表）"""
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    description: Optional[str] = None
    genre: Optional[str] = None
    target_word_count: int
    current_word_count: int
    status: str
    outline: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    chapters: List[ChapterSummary] = []


class ProjectListResponse(BaseModel):
    """项目列表响应（不含章节内容）"""
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    description: Optional[str] = None
    genre: Optional[str] = None
    target_word_count: int
    current_word_count: int
    status: str
    outline: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
